Code/Framework/AzToolsFramework/AzToolsFramework/ViewportUi/ButtonGroup.cpp

```diff
@@ -57,7 +57,15 @@ namespace AzToolsFramework::ViewportUi::Internal
 
     ButtonId ButtonGroup::AddButton(const AZStd::string& icon, const AZStd::string& name)
     {
-        auto buttonId = ButtonId(m_buttons.size() + 1);
+        auto max = ButtonId(0);
+        for (const auto& button : m_buttons)
+        {
+            if (button.first > max)
+            {
+                max = button.first;
+            }
+        }
+        auto buttonId = ButtonId(max + 1);
```

Comment:
This was changed because say you had a switcher with 3 buttons: 1, 2, 3. If the button  with ID: 2 is deleted, there are two buttons left in the switcher. With the previous method, it would create a new button with ID: 3, which conflicts with the existing buttons. This worked previously because the buttons were never technically deleted from the button group and infinite buttons would be created. I changed that below in this file, where RemoveButton actually removes the button from the button group.

Reply:
Ah good find! 🙂